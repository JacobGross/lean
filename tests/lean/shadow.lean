open nat

variable a : nat

-- The variable 'a' in the following definition is not the variable 'a' above
definition tadd : nat → nat → nat
| tadd zero     b := b
| tadd (succ a) b := succ (tadd a b)
