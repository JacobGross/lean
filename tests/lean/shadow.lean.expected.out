shadow.lean:8:13: error: invalid recursive equation, variable 'a' has the same name of a variable in an outer-scope (solution: rename this variable)
